webpack.config.js

```diff
@@ -0,0 +1,85 @@
+// var ConcatSource = require("webpack/lib/ConcatSource");
+var BannerPlugin = require("webpack/lib/BannerPlugin");
+var fs = require("fs");
+var path = require("path");
+var webpack = require("webpack");
+
+var distDir = path.join(__dirname, "dist");
+
+module.exports = [
+  // Plain build.
+  {
+    entry: "./lib/source-map.js",
+    output: {
+      path: distDir,
+      filename: "source-map.js",
+      library: "sourceMap",
+      libraryTarget: "umd",
+    },
+  },
+
+  // Debug build.
+  {
+    entry: "./lib/source-map.js",
+    output: {
+      path: distDir,
+      filename: "source-map.debug.js",
+      library: "sourceMap",
+      libraryTarget: "umd",
+    },
+    devtool: "#inline-source-map"
+  },
+
+  // Minified build.
+  {
+    entry: "./lib/source-map.js",
+    output: {
+      path: distDir,
+      filename: "source-map.min.js",
+      library: "sourceMap",
+      libraryTarget: "umd",
+    },
+    plugins: [
+      new webpack.optimize.UglifyJsPlugin({
+        sourceMap: true
+      })
+    ],
+    devtool: "#source-map"
+  }
+];
+
+/*** Build the tests for inclusion in mozilla-central. ************************/
+
+var testFileRegex = /^test\-.*?\.js/;
+
+function isTestFile(file) {
+  return testFileRegex.test(file);
+}
+
+var testsDir = path.join(__dirname, "test", "source-map");
+var testFiles = fs.readdirSync(testsDir).filter(isTestFile);
+
+// The xpcshell test harness expects a run_test function. Define this function
+// and stringify it as banner at the top of each test file.
+function run_test() {
+  for (var k in SOURCE_MAP_TEST_MODULE) {
+    if (/^test/.test(k)) {
+      SOURCE_MAP_TEST_MODULE[k](assert);
+    }
+  }
+}
+
+testFiles.forEach(function (file) {
+  module.exports.push({
```

Comment:
Yeah some duplication, but only in tests. The build process is just so much more sane than it used to be, that I think this is worth it.